scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -166,46 +177,74 @@ final case class SpecificRecordIO[T <: SpecificRecord: ClassTag: Coder](path: St
     val cls = ScioUtil.classOf[T]
     val t = beam.AvroIO.write(cls)
     data.applyInternal(
-      avroOut(data, t, path, params.numShards, params.suffix, params.codec, params.metadata)
+      avroOut(data, t, paths.head, params.numShards, params.suffix, params.codec, params.metadata)
     )
     tap(())
   }
 
   override def tap(read: ReadP): Tap[T] =
-    SpecificRecordTap[T](ScioUtil.addPartSuffix(path))
+    SpecificRecordTap[T](ScioUtil.addPartSuffix(paths.head))
 }
 
-final case class GenericRecordIO(path: String, schema: Schema) extends AvroIO[GenericRecord] {
+object SpecificRecordIO {
+  def apply[T <: SpecificRecord: ClassTag: Coder](path: String): SpecificRecordIO[T] =
+    SpecificRecordIO(Iterable(path))
+}
+
+final case class GenericRecordIO(paths: Iterable[String], schema: Schema)
+    extends AvroIO[GenericRecord] {
   override type ReadP = Unit
   override type WriteP = AvroIO.WriteParam
 
-  override def testId: String = s"AvroIO($path)"
+  // TODO: support test with multiple paths
+  override def testId: String = s"AvroIO(${paths.head})"
```

Comment:
Does not seem super safe.

Reply:
yes, we need to improve to handle multiple files, it is one of the todos in the PR. This is just to get all existing tests to pass.